### PR TITLE
New bid adapter for TheAdx

### PR DESCRIPTION
**Pull request by mustafakemal16**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

New bid adapter for TheAdx

- test parameters for validating bids
```
   var adUnits = [{
     code: 'div-gpt-ad-1460505748561-0',
     mediaTypes: {
         banner: {
             sizes: [
                 [300, 250],
                 [300, 600]
             ],
         }
     },
     bids: [{
         bidder: 'theadx',
         params: {
             pid: 1, // publisher id
             wid: 7, // website id
             tagId: 18, //zone id
             url: window.location.href || window.top.location.href
         }
     }]

 }];


```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer 

prebid@theadx.com

- [X ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/mustafakemal16/prebid.github.io

### CODE REVIEW

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/1150

**Comment by mike-chowla** [comment]
The test parameters are not getting a bid back. 
I'm seeing a request go out to http://ssp.theadx.com/request?tagid=18 but a zero length response is coming back

File: modules/theAdxBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    let res = false;
+    if (bid && bid.params) {
+      res = !!(bid.params.pid && bid.params.tagId && bid.params.url);
```
Comment (mike-chowla) [inline modules/theAdxBidAdapter.js]: You should require url to be a bidder parameter.  Please use bidderRequest.refererInfo instead
See http://prebid.org/dev-docs/bidder-adaptor.html#referrers

File: modules/theAdxBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+      });
+    }
+
+    return syncs;
+  },
+
+  /**
+   * @param {TimedOutBid} timeoutData
+   */
+  onTimeout: (timeoutData) => {
```
Comment (mike-chowla) [inline modules/theAdxBidAdapter.js]: I'm not clear on how this is being called.

File: modules/theAdxBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+        related_data: timeoutData[0].timeout || config.getConfig('bidderTimeout')
+      };
+      logEvent(eventData, timeoutData);
+    } catch (e) {}
+  },
+
+  /**
+   * @param {TimedOutBid} timeoutData
+   */
+  onBidWon: (bid) => {
```
Comment (mike-chowla) [inline modules/theAdxBidAdapter.js]: I'm not clear on how this function is being called.   Could you elaborate?

File: modules/theAdxBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+  return params;
+}
+
+function logEvent (event, data) {
+  let getParams = {
+    protocol: 'https',
+    hostname: EVENT_PIXEL_URL,
+    search: getLoggingData(event, data)
+  };
+  utils.triggerPixel(url.format(getParams));
```
Comment (mike-chowla) [inline modules/theAdxBidAdapter.js]: Adapters are prohibited from firing off additional network calls.

File: modules/theAdxBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+}
+
+let buildSiteComponent = (bidRequest) => {
+  let topLocation = utils.getTopWindowLocation();
```
Comment (mike-chowla) [inline modules/theAdxBidAdapter.js]: You shoulde be using bidderRequest.refererInfo here as getTopWindownLocation() is deprecated

File: modules/theAdxBidAdapter.js (hunk 1 above)
Comment (mustafakemal16) [reply modules/theAdxBidAdapter.js]: url removed, we currently use refererInfo.

File: modules/theAdxBidAdapter.js (hunk 5 above)
Comment (mustafakemal16) [reply modules/theAdxBidAdapter.js]: onTimeout&onBidWon events removed, so its not necessary,removed

File: modules/theAdxBidAdapter.js (hunk 4 above)
Comment (mustafakemal16) [reply modules/theAdxBidAdapter.js]: onTimeout&onBidWon events removed, so its not necessary,removed

**Comment by mustafakemal16** [comment]
> The test parameters are not getting a bid back.
> I'm seeing a request go out to http://ssp.theadx.com/request?tagid=18 but a zero length response is coming back

Hi Mike,
i think, its a domain issue, which domain call to test page? we add test.localhost support. localhost && test.localhost supported in this tag.
best regards

**Comment by mike-chowla** [comment]
I was able to get a bid using localhost.

**Pull request merged**